x-pack/plugins/transform/public/app/sections/create_transform/components/step_details/step_details_form.tsx

```diff
@@ -93,9 +118,31 @@ export const StepDetailsForm: FC<Props> = React.memo(
     );
     const [transformIds, setTransformIds] = useState<TransformId[]>([]);
     const [indexNames, setIndexNames] = useState<EsIndexName[]>([]);
+
+    // Index pattern state
     const [indexPatternTitles, setIndexPatternTitles] = useState<IndexPatternTitle[]>([]);
     const [createIndexPattern, setCreateIndexPattern] = useState(defaults.createIndexPattern);
+    const [previewDateColumns, setPreviewDateColumns] = useState<string[]>([]);
+    const [indexPatternDateField, setIndexPatternDateField] = useState<string | undefined>(
+      undefined
```

Comment:
nit: no need to explicitly set `undefined`

Reply:
Fixed in https://github.com/elastic/kibana/pull/68842/commits/3fe458e73e36261939a364c1a1101626b85ba7b1